test_ros2cli/test/test_cli.py.in

```diff
@@ -0,0 +1,139 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import sys
+
+import unittest
+
+from launch import LaunchDescription
+from launch.actions import ExecuteProcess
+from launch.actions import OpaqueFunction
+from launch.actions import TimerAction
+from launch.substitutions import LaunchConfiguration
+
+import launch_testing
+import launch_testing.asserts
+import launch_testing.util
+
+sys.path.append(os.path.dirname(__file__))
+
+from @TEST_CONFIG_MODULE@ import get_test_configurations  # noqa
+
+
+@launch_testing.parametrize(
+    'test_configuration', get_test_configurations(
+        rmw_implementation='@TEST_RMW_IMPLEMENTATION@'
+    )
+)
+def generate_test_description(test_configuration, ready_fn):
+    cmd = ['ros2', test_configuration.command]
+    cmd.extend(test_configuration.arguments)
+    command_under_test = ExecuteProcess(
+        cmd=cmd,
+        name='cli',
+        output='screen',
+        additional_env={'PYTHONUNBUFFERED': '1'},
+        sigterm_timeout=LaunchConfiguration(
+            'sigterm_timeout', default=60
+        )
+    )
+    actions = [command_under_test]
+    actions.append(
+        OpaqueFunction(function=lambda context: ready_fn())
+    )
+    launch_description = LaunchDescription([
+        # Always restart daemon to isolate tests.
+        ExecuteProcess(
+            cmd=['ros2', 'daemon', 'stop'],
+            name='daemon-stop',
+            on_exit=[
+                ExecuteProcess(
+                    cmd=['ros2', 'daemon', 'start'],
+                    name='daemon-start',
+                    on_exit=[
+                        # Add test fixture actions.
+                        *test_configuration.fixture_actions,
+                        # Spin up command under test with a setup delay.
+                        TimerAction(
+                            period=test_configuration.setup_delay or @TEST_SETUP_DELAY@,
+                            actions=actions,
+                        )
+                    ]
+                )
+            ]
+        ),
+    ])
+    return launch_description, locals()
+
+
+class TestCommandFinishesInAFiniteAmountOfTime(unittest.TestCase):
+
+    def @TEST_NAME@(self, proc_info, command_under_test, test_configuration):
+        """Test that the command under test finished in a finite amount of time."""
+        success = proc_info.waitForShutdown(
+            process=command_under_test, timeout=test_configuration.timeout
```

Comment:
`waitForShutdown` could take an optional keyword argument: `assert: bool = False`. In that way, we wouldn't need the extra `if ...: assert ...` here.
I think it may be useful.
Note: We have also `assertWaitForShutdown`, but it's less versatile than an extra argument.
Note: I won't delete the original `waitForShutdown`